File: cpp/blind curated 75/contains_duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

bool containsDuplicate(vector<int> &nums)
{
    unordered_set<int> unique;

    for (int i = 0; i < nums.size(); i++)
    {
        if (unique.find(nums.at(i)) != unique.end())
        {
            return true;
        }

        unique.insert(nums.at(i));
    }

    return false;
}

int main()
{
    vector<int> v1{1, 2, 3, 1};
    cout << containsDuplicate(v1) << endl; // true
}

File: python/blind curated 75/contains_duplicate.py
# Leetcode 217


def containsDuplicate(nums: list) -> bool:
    unique = set()

    for i in nums:
        if i in unique:
            return True
        unique.add(i)

    return False


print(containsDuplicate([1, 2, 3, 1]))  # True
print(containsDuplicate([1, 2, 3]))  # False
